vunit/ui/results.py

```diff
@@ -24,5 +29,62 @@ def merge_coverage(self, file_name, args=None):
         :param file_name: The resulting coverage file name.
         :param args: The tool arguments for the merge command. Should be a list of strings.
         """
-
         self._simulator_if.merge_coverage(file_name=file_name, args=args)
+
+    def get_report(self):
+        """
+        Get a report (dictionary) of tests: status, time and output path
+        """
+
+        def get_relative_test_path(test_path):
+            """
+            If the test_path is relative to the default TEST_OUTPUT_PATH, return the subdir only
+            """
+            base = basename(test_path)
+            return (
+                base
+                if normpath(join(self._output_path, TEST_OUTPUT_PATH, base))
+                == normpath(test_path)
+                else test_path
+            )
+
+        report = Report(self._output_path)
+        for (
+            test
+        ) in self._report._test_results_in_order():  # pylint: disable=protected-access
+            obj = test.to_dict()
+            report.tests.update(
+                {
+                    test.name: ReportResult(
+                        obj["status"], obj["time"], get_relative_test_path(obj["path"])
+                    )
+                }
+            )
+        return report
+
+
+class Report(object):
+    """
+    Gives access to test results and paths after running tests
+    """
+
+    def __init__(self, output_path):
+        self.output_path = output_path
+        self.tests = {}
+
+    def get_full_path(self, test_path, file_path=""):
```

Comment:
This method belongs on the individual test result (ReportResult) class.

Reply:
See https://github.com/VUnit/vunit/pull/578#discussion_r345582338